File: src/classes/ha_comm/set_auto_off.rs
use crate::utils::byte_arrays_helper::ByteArraysHelper;

pub struct SetAutoOff{
    off_timeout: u8,
}

impl SetAutoOff{
    pub fn new() -> Self{
        Self{
            off_timeout: 0,
        }
    }

    pub fn get_bytes(&self) -> Vec<u8>{
        let (b1, b2) = ByteArraysHelper::word_to_port16(self.off_timeout as u16);
        
        return ByteArraysHelper::combine_2b(b1, b2);
    }
}

File: lib/classes/hacommand/set_auto_off.py
from lib.utils.byte_arrays_helper import ByteArraysHelper


class SetAutoOff:
    def __init__(self):
        self.off_timeout = 0

    def get_bytes(self):
        b1, b2 = ByteArraysHelper.word_to_port16(self.off_timeout)
        return ByteArraysHelper.combine_2b(b1, b2)

    def set_bytes(self, data):
        self.off_timeout = ByteArraysHelper.port16_to_word(data)
        